gateway/handle.go

```diff
@@ -614,3 +614,228 @@ func (gw *Handle) addToWebRequestQ(_ *http.ResponseWriter, req *http.Request, do
 	}
 	userWebRequestWorker.webRequestQ <- &webReq
 }
+
+func (gw *Handle) writeToJobsDB() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		reqType := ctx.Value(gwtypes.CtxParamCallType).(string)
+		arctx := ctx.Value(gwtypes.CtxParamAuthRequestContext).(*gwtypes.AuthRequestContext)
+
+		// TODO: add tracing
+		gw.logger.LogRequest(r)
+		var errorMessage string
+		defer func() {
+			if errorMessage != "" {
```

Comment:
This approach will become hard to maintain given the size of the function:
* It is much easier to test for errors when the method returns error, instead of writing in an http response. (similar intention with my comment above)
* we need to remember to set `errorMessage` when returning with an error.

Reply:
I've split the function quite a bit, so this looks a little better. Let me know after you take a look if we still need a little more work on it..